lib/improver/tests/blending/weights/test_WeightsUtilities.py

```diff
@@ -140,6 +140,23 @@ def test_returns_correct_values(self):
         expected_result = np.array([0.6, 0.3, 0.1])
         self.assertArrayAlmostEqual(result, expected_result)
 
+    def test_returns_correct_values_2darray_axis0(self):
+        """Test normalizing along the collumns of the array."""
+        weights_in = np.array([[6.0, 3.0, 1.0],
+                               [4.0, 1.0, 3.0]])
+        result = WeightsUtilities.normalise_weights(weights_in, axis=0)
+        expected_result = np.array([[0.6, 0.75, 0.25],
+                                    [0.4, 0.25, 0.75]])
+        self.assertArrayAlmostEqual(result, expected_result)
+
+    def test_returns_correct_values_2darray_axis1(self):
+        """Test normalizing along the rows of the array."""
+        weights_in = np.array([[6.0, 3.0, 1.0],
+                               [4.0, 1.0, 3.0]])
+        result = WeightsUtilities.normalise_weights(weights_in, axis=1)
+        expected_result = np.array([[0.6, 0.3, 0.1],
+                                    [0.5, 0.125, 0.375]])
+        self.assertArrayAlmostEqual(result, expected_result)
```

Comment:
Can you add a test for the case we are trying to solve - correct behaviour when one of the elements is zero.

Reply:
Done